File: src/main/java/xyz/lncvrt/galaxyboxpvp/events/FurnaceSmeltListener.java
package xyz.lncvrt.galaxyboxpvp.events;

import xyz.lncvrt.galaxyboxpvp.GalaxyBoxPvP;
import org.bukkit.Material;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.FurnaceSmeltEvent;

public class FurnaceSmeltListener implements Listener {
    private final GalaxyBoxPvP plugin;

    public FurnaceSmeltListener(GalaxyBoxPvP plugin) {
        this.plugin = plugin;
    }

    @EventHandler
    public void onFurnaceSmelt(FurnaceSmeltEvent event) {
        Material smelted = event.getSource().getType();

        if (plugin.isRestrictedMaterial(smelted)) {
            event.setCancelled(true);
        }
    }
}

File: src/main/kotlin/xyz/lncvrt/galaxyboxpvp/events/FurnaceSmeltListener.kt
package xyz.lncvrt.galaxyboxpvp.events

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.FurnaceSmeltEvent
import xyz.lncvrt.galaxyboxpvp.GalaxyBoxPvPPlugin

class FurnaceSmeltListener(private val plugin: GalaxyBoxPvPPlugin) : Listener {
    @EventHandler
    fun onFurnaceSmelt(event: FurnaceSmeltEvent) {
        val smelted = event.source.type
        if (plugin.isRestrictedMaterial(smelted)) event.isCancelled = true
    }
}